cogboard-app/src/main/kotlin/com/cognifide/cogboard/http/HttpClient.kt

```diff
@@ -86,18 +86,25 @@ class HttpClient : AbstractVerticle() {
     private fun initRequest(request: HttpRequest<Buffer>, config: JsonObject): HttpRequest<Buffer> {
         val user = config.getString(CogboardConstants.PROP_USER) ?: ""
         val pass = config.getString(CogboardConstants.PROP_PASSWORD) ?: ""
+        val token = config.getString(CogboardConstants.PROP_TOKEN) ?: ""
         val headers = config.getJsonObject(CogboardConstants.PROP_HEADERS)
 
-        if (user.isNotBlank() && pass.isNotBlank()) {
+        if (user.isNotBlank() && token.isNotBlank()) {
+            request.basicAuthentication(user, token)
+        } else if (user.isNotBlank() && pass.isNotBlank()) {
             request.basicAuthentication(user, pass)
-            request.putHeader("Content-Type", "application/json")
         }
 
+        setRequestHeaders(request, headers)
+
+        return request
+    }
+
+    private fun setRequestHeaders(request: HttpRequest<Buffer>, headers: JsonObject?) {
```

Comment:
please use `apply` instead of `set` keyword

Reply:
when I see `set...` it looks for me like we are changing the state of `this` object.

Reply:
fixed